Content.Shared/_Shitcode/Heretic/Systems/RiposteeSystem.cs

```diff
@@ -101,72 +116,128 @@ private void OnHarmAttempt(Entity<RiposteeComponent> ent, ref BeforeHarmfulActio
         if (args.Cancelled)
             return;
 
-        if (!ent.Comp.CanRiposte)
+        if (_net.IsClient)
             return;
 
-        if (_net.IsClient)
+        if (_mobState.IsIncapacitated(ent))
             return;
 
-        if (args.User == ent.Owner)
+        if (HasComp<RiposteeComponent>(args.User))
             return;
 
-        Entity<MeleeWeaponComponent>? weapon = null;
-        foreach (var held in _hands.EnumerateHeld(ent.Owner))
+        foreach (var data in ent.Comp.Data.Values)
         {
-            if (_whitelist.IsWhitelistPassOrNull(ent.Comp.WeaponWhitelist, held) &&
-                TryComp(held, out MeleeWeaponComponent? melee))
-                weapon = (held, melee);
-        }
+            if (!data.CanRiposte)
+                continue;
 
-        if (weapon == null)
-            return;
+            if (args.User == ent.Owner)
+                continue;
 
-        if (!_blocker.CanAttack(ent, args.User, weapon.Value))
-            return;
+            if (data.CanRiposteEvent != null)
+            {
+                var ev = data.CanRiposteEvent;
+                ev.Handled = false;
+                RaiseLocalEvent(ent, (object) ev);
+                if (!ev.Handled)
+                    continue;
+            }
 
-        args.Cancel();
+            if (!data.CanRiposteWhileProne && _standing.IsDown(ent))
+                continue;
 
-        CounterAttack(weapon.Value, ent, args.User);
-        RaiseNetworkEvent(new RiposteUsedEvent(GetNetEntity(ent.Owner),
-                GetNetEntity(args.User),
-                GetNetEntity(weapon.Value.Owner)),
-            ent.Owner);
+            if (data.RiposteChance is > 0f and < 1f && !_random.Prob(data.RiposteChance))
```

Comment:
What the fuck is this mix of conditionals lmao
```suggestion
            if (data.RiposteChance is > 0f and < 1f
                && !_random.Prob(data.RiposteChance))